lib/elixir/lib/uri.ex

```diff
@@ -666,6 +666,9 @@ defmodule URI do
             end
```

Comment:
I think the branch line 664 could also lead to an `:undefined` port?

Reply:
Absolutely right! I was able to get reproduce using a scheme which has no port mapped through default_port/1 and added a test for it in https://github.com/elixir-lang/elixir/pull/13464/commits/ed85d094c44ca637e400816fbf83bc98cd2b8af9

Reply:
Wondering if anything speaks against simplifying the above to:

```
%{port: port} when is_integer(port) ->
  %{uri | scheme: scheme}
  
%{} ->
  %{uri | scheme: scheme, port: default_port(scheme)}
```